server/command.go

```diff
@@ -145,6 +147,20 @@ func (p *Plugin) ExecuteCommand(c *plugin.Context, args *model.CommandArgs) (*mo
 	ctx, cancel := context.WithTimeout(context.Background(), commandTimeout)
 	defer cancel()
 
+	defer func() {
+		if r := recover(); r != nil {
+			p.client.Log.Warn("Recovered from a panic",
+				"Command", args.Command,
+				"UserId", args.UserId,
+				"error", r,
+				"stack", string(debug.Stack()))
+			err = model.NewAppError("ExecuteCommand", "plugin_gitlab.command.execute_command.panic", nil, "", http.StatusInternalServerError)
+			res = &model.CommandResponse{}
+			res.Text = "An unexpected error occurred. Please try again later."
+			res.Username = p.BotUserID
```

Comment:
One improvement comes to mind here. This current approach may work, though not sure if it actually posts _with_ bot account itself (does the GitLab bot icon show in the response?), rather than some username override thing happening here. We can have the Gitlab bot respond using the following code:

```suggestion
			res = &model.CommandResponse{}
			p.postCommandResponse(args, "An unexpected error occurred. Please try again later.")
```

Note that this pattern (and bot accounts themselves) emerged in plugins a while after the plugin framework was created. It is a bit manual, though there is no inherent coupling to the command response and the bot associated with the plugin (plugins can potentially have multiple bots associated with them). This approach allows the user to know that the issue happened with the GitLab plugin specifically.